source/AI.cpp

```diff
@@ -651,12 +651,33 @@ void AI::Step(const PlayerInfo &player, Command &activeCommands)
 			continue;
 		}
 
-		// Run away if your hostile target is not disabled and you are badly damaged.
+		// Check if a ship is armed with only ammo-using weapons,
+		// but no longer has the ammunition needed to use them.
+		bool needsAmmo = false;
+		for(const Hardpoint &hardpoint : it->Weapons())
+		{
+			const Weapon *weapon = hardpoint.GetOutfit();
+			if(weapon && !hardpoint.IsAntiMissile())
+			{
+				const Outfit *ammo = weapon->Ammo();
+				if(!ammo || it->OutfitCount(ammo))
+				{
+					// This ship has at least one usable weapon.
+					needsAmmo = false;
+					break;
+				}
+				else
+					needsAmmo = true;
+			}
+		}
```

Comment:
You should determine this in `ShipAICache::Calibrate()` and provide a method in that class to retrieve the result.

Reply:
Is it OK now?